cli/src/registry/publish.ts

```diff
@@ -0,0 +1,227 @@
+import { execSync } from 'child_process';
+import { createCommand } from 'commander';
+import { rmSync } from 'fs';
+import { writeFile } from 'fs/promises';
+import { tmpdir } from 'os';
+import { join } from 'path';
+import { AddRegistryEntryResult, EditRegistryEntryResult } from '../generated/control_panel';
+import { cargoProjectVersion, execAsync, readFileIntoUint8Array, toBlobString } from '../utils';
+import {
+  Application,
+  applicationToRegistryEntryMap,
+  parseRegistryApplication,
+  registryEntryToApplicationMap,
+  searchRegistry,
+} from './registry.core';
+
+const artifactsRootPath = join(__dirname, '../../../artifacts');
+
+interface PackageDedependency {
+  name: string;
+  version: string;
+}
+
+interface Package {
+  name: string;
+  description: string;
+  tags: string[];
+  categories: string[];
+  metadata: Record<string, string>;
+  version: string;
+  module: Uint8Array;
+  dependencies: PackageDedependency[];
+}
+
+const getAppRegistryEntry = async (app: Application, network: string): Promise<Package> => {
```

Comment:
There's no default cause in the switch statement so the correct return type should be `Promise<Package | undefined>`. Maybe a simple `throw` would be nice to help debug it in case it somehow goes wrong. I think it's fine though since it's just a CLI for local use.

Reply:
Typescript is still happy because `Application` has only 2 variants and both are covered in the switch, but just to be on the safe side i added the default with a throw as you suggested.